src/Core/Content/Product/SalesChannel/Review/ProductReviewLoader.php

```diff
@@ -0,0 +1,170 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Content\Product\SalesChannel\Review;
+
+use Shopware\Core\Content\Product\Aggregate\ProductReview\ProductReviewEntity;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Aggregation\Bucket\FilterAggregation;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Aggregation\Bucket\TermsAggregation;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\AggregationResult\Bucket\TermsResult;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\EqualsFilter;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\MultiFilter;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\RangeFilter;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Sorting\FieldSorting;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Routing\Exception\MissingRequestParameterException;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+#[Package('inventory')]
+class ProductReviewLoader
+{
+    private const LIMIT = 10;
+    private const DEFAULT_PAGE = 1;
+    private const FILTER_LANGUAGE = 'filter-language';
+    private const DEFAULT_SORTING = 'createdAt';
+    private const ALLOWED_SORTINGS = ['createdAt', 'points'];
+
+    /**
+     * @internal
+     */
+    public function __construct(
+        private readonly AbstractProductReviewRoute $route,
+        private readonly EventDispatcherInterface $eventDispatcher
+    ) {
+    }
+
+    /**
+     * load reviews for one product. The request must contain the productId
+     * otherwise MissingRequestParameterException is thrown
+     *
+     * @throws MissingRequestParameterException
+     * @throws \Shopware\Core\Framework\DataAbstractionLayer\Exception\InconsistentCriteriaIdsException
+     */
+    public function load(Request $request, SalesChannelContext $context): ProductReviewLoaderResult
```

Comment:
can you add a unit test for this one? 

and please also add small unit tests fir the result struct and event

Reply:
I added the tests for the result struct and event.

Regarding the unit test of the loader itself, should every code branch be tested? If you have an example in mind with a comparable test, it would probably help, as I am not really sure how extensive this test should be.

Reply:
100% coverage should always be the goal :see_no_evil: 

but from glancing at the code i think we should also add at least one integration test to ensure all the stuff is handled correctly as this class is mainly creating criterias for the repo, having a unit test that mocks the repo does not provide that much value

Reply:
@keulinho I now added an unit test for the `ProductReviewLoader`. Regarding the integration test, I checked and I am not sure if one wants to expose this functionality to the storefront API and add an integration test for this route, or if one should write an integration test for this "loader".

With this change the `ProductReviewRoute` is not used anymore in the core, but rather the loader is used everywhere, so I am not sure if one wants to change the route instead of adding this loader: https://github.com/shopware/platform/blob/174f160d1930f995e63516c7bae0282dafa1d668/src/Core/Content/Product/SalesChannel/Review/ProductReviewRoute.php#L31-L32
And I if so, I am not sure how to keep the backwards compatibility, properly for Store API routes.

Reply:
Sorry when i was not clear here, what i meant was to add integration test specifically and directly for this loader, 
really cool and nice that you added unit tests for this class :+1: 

i just would like to see one basic happy path test that really adds a product with 2 reviews and checks that the sorting and aggregation really works as expected

So you would test this class directly with phpunit and not over the store api 

Is that clearer for you now?

Reply:
So :-) Finally I added the integration test as well :-)